homeassistant/components/airzone/binary_sensor.py

```diff
@@ -0,0 +1,115 @@
+    DEVICE_CLASS_PROBLEM,
+    DEVICE_CLASS_RUNNING,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import AirzoneEntity
+from .const import DOMAIN
+from .coordinator import AirzoneUpdateCoordinator
+
+
+@dataclass
+class AirzoneBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes airzone binary sensor entities."""
+
+    attributes: dict[str, str] | None = None
+
+
+BINARY_SENSOR_TYPES: Final[tuple[AirzoneBinarySensorEntityDescription, ...]] = (
+    AirzoneBinarySensorEntityDescription(
+        attributes={
+            "air": AZD_AIR_DEMAND,
+            "floor": AZD_FLOOR_DEMAND,
+        },
+        device_class=DEVICE_CLASS_RUNNING,
+        key=AZD_DEMAND,
+        name="Demand",
+    ),
+    AirzoneBinarySensorEntityDescription(
+        attributes={
+            "errors": AZD_ERRORS,
+        },
+        device_class=DEVICE_CLASS_PROBLEM,
+        key=AZD_PROBLEMS,
+        name="Problem",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Airzone binary sensors from a config_entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    binary_sensors = []
+    for system_zone_id, zone_data in coordinator.data[AZD_ZONES].items():
+        for description in BINARY_SENSOR_TYPES:
+            if description.key in zone_data:
+                binary_sensors.append(
+                    AirzoneBinarySensor(
+                        coordinator,
+                        description,
+                        entry,
+                        system_zone_id,
+                        zone_data,
+                    )
+                )
+
+    async_add_entities(binary_sensors)
+
+
+class AirzoneBinarySensor(AirzoneEntity, BinarySensorEntity):
+    """Define an Airzone sensor."""
+
+    def __init__(
+        self,
+        coordinator: AirzoneUpdateCoordinator,
+        description: AirzoneBinarySensorEntityDescription,
+        entry: ConfigEntry,
+        system_zone_id: str,
+        zone_data: dict[str, Any],
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator, entry, system_zone_id, zone_data)
+        self._attr_name = f"{zone_data[AZD_NAME]} {description.name}"
+        self._attr_unique_id = f"{entry.entry_id}_{system_zone_id}_{description.key}"
+        self.entity_description = description
+        self.attributes = description.attributes
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return state attributes."""
+        _state_attr = None
+        if self.attributes:
+            _state_attr = {}
+            for key, val in self.attributes.items():
+                _state_attr[key] = self.get_zone_value(val)
```

Comment:
This should be a dict comprehension. 
```
_state_attr = {
  key: self.get_zone_value(val)
  for key, val in self.attributes.items()
}
```

Reply:
Sure, excuse me but my python skills are a bit limited >.<
Changed in 7db4c8dabb4c2cfb0002cf4259892e797ed58a86

Reply:
That is fine. This is a minor suggestion, that somebody else might make later. You probably should do an early return here.

```
if not self.attributes:
  return None

return { dict comprehension 
```

Reply:
OK!
Changed in 213edc291a83b227410b10a7aa25c23b3166ccdc